test/Microsoft.TestPlatform.Extensions.TrxLogger.UnitTests/TrxLoggerTests.cs

```diff
@@ -634,6 +639,95 @@ public void DefaultTrxFileShouldCreateIfLogFileNameParameterNotPassed()
             Assert.IsFalse(string.IsNullOrWhiteSpace(this.testableTrxLogger.trxFile));
         }
 
+        [TestMethod]
+        public void DefaultTrxFileNameVerification()
+        {
+            this.parameters.Remove(TrxLoggerConstants.LogFileNameKey);
+            this.parameters[TrxLoggerConstants.LogFilePrefixKey] = DefaultLogFilePrefixParameterValue;
+
+            var time = DateTime.Now;
+            var trxFileHelper = new TrxFileHelper(() => time);
+
+            testableTrxLogger = new TestableTrxLogger(new FileHelper(), trxFileHelper);
+            testableTrxLogger.Initialize(this.events.Object, this.parameters);
+
+            MakeTestRunComplete();
+
+            var fileName = Path.GetFileName(testableTrxLogger.trxFile);
+            var expectedName = $"{DefaultLogFilePrefixParameterValue}{time:_yyyyMMddHHmmss}.trx";
+
+            Assert.AreEqual(expectedName, fileName, "Trx file name pattern has changed. It should be in the form of prefix_yyyyMMddHHmmss.trx, Azure Devops VSTest task depends on this naming.");
+        }
+
+        [TestMethod]
+        public void DefaultTrxFileShouldIterateIfLogFileNameParameterNotPassed()
+        {
+            this.parameters.Remove(TrxLoggerConstants.LogFileNameKey);
+
+            var files = TestMultipleTrxLoggers();
+
+            Assert.IsTrue(files.Length == MultipleLoggerInstanceCount, "All logger instances should get a seperate file name!");
+        }
+
+        [TestMethod]
+        public void TrxFileNameShouldNotIterate()
+        {
+            var files = TestMultipleTrxLoggers();
+
+            Assert.IsTrue(files.Length == 1, "All logger instances should get the same file name!");
+        }
+
+        [TestMethod]
+        public void TrxPrefixFileNameShouldIterate()
+        {
+            this.parameters.Remove(TrxLoggerConstants.LogFileNameKey);
+            this.parameters[TrxLoggerConstants.LogFilePrefixKey] = DefaultLogFilePrefixParameterValue;
+
+            var files = TestMultipleTrxLoggers();
+
+            Assert.IsTrue(files.Length == MultipleLoggerInstanceCount, "All logger instances should get the same file name!");
```

Comment:
One more nitpick, use equality assertions instead so we can see the actual numbers in the failure. Also is the description correct? it is the same for the positive and negative case.

Reply:
Updated the assertions. For that configuration of instances, it's true. Negative case is also tested in other method.